job-server/src/main/scala/spark/jobserver/AkkaClusterSupervisorActor.scala

```diff
@@ -152,8 +155,14 @@ class AkkaClusterSupervisorActor(daoActor: ActorRef) extends InstrumentedActor {
         logger.info("Shutting down context {}", name)
         val contextActorRef = contexts(name)._1
         cluster.down(contextActorRef.path.address)
-        contextActorRef ! PoisonPill
-        sender ! ContextStopped
+        try {
+          val stoppedCtx = gracefulStop(contexts(name)._1, 2 seconds)
```

Comment:
Done with the latest commit. Parameters are:
spark.jobserver.yarn-context-deletion-timeout and spark.jobserver.context-deletion-timeout